### ISSUE TITLE
What is the proposal AP supposed to measure?

### ISSUE THREAD

**Issue by snailrowen1337** [facebookresearch/detectron2#3325]
I have trained a maskr-rcnn model on coco and would like to evaluate its recall. On the bbox metric, I get recalls like

```
Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.332  
Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.524 
Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.551     
```

I also want to evaluate the recall of the proposals by using the code [here](https://github.com/facebookresearch/detectron2/blob/master/detectron2/evaluation/coco_evaluation.py#L252). To do this I simply modify the inference function of mask-r-cnn:

```
    def inference(
        self,
        batched_inputs,
        detected_instances=None,
        do_postprocess=True,
    ):


        assert not self.training

        images = self.preprocess_image(batched_inputs)
        features = self.backbone(images.tensor)

        if detected_instances is None:
            if self.proposal_generator is not None:
                proposals, _ = self.proposal_generator(images, features, None)
            else:
                assert "proposals" in batched_inputs[0]
                proposals = [x["proposals"].to(self.device) for x in batched_inputs]

            results, _ = self.roi_heads(images, features, proposals, None)
        else:
            detected_instances = [x.to(self.device) for x in detected_instances]
            results = self.roi_heads.forward_with_given_boxes(features, detected_instances)

        if do_postprocess:
            assert not torch.jit.is_scripting(), "Scripting is not supported for postprocess."
            results = GeneralizedRCNN._postprocess(results, batched_inputs, images.image_sizes)

        # evaluate proposals too
        for rdict, props in zip(results, proposals):
            rdict['proposals'] = props
        return results
```

However, doing this gives me incredibly poor recall for the proposals, so there must be a bug somewhere:

```
|  AR@100  |  ARs@100  |  ARm@100  |  ARl@100  |  AR@1000  |  ARs@1000  |  ARm@1000  |  ARl@1000  | 
| 1.456   |   0.029   |   0.541   |   5.272   |   5.490   |   0.263    |   3.109    |   18.084   |  
```

How is the proposal recall in [_eval_box_proposals](https://github.com/facebookresearch/detectron2/blob/master/detectron2/evaluation/coco_evaluation.py#L252) in coco_evaluation.py supposed to be used? This does not seem to be documented.Thanks!!
